### ISSUE TITLE
File name too long when downloading models

### ISSUE THREAD

**Issue by Hxyou** [huggingface/huggingface_hub#1426]
### Describe the bug

When I am downloading a pre-trained model, the following error occurs: 
`OSError: [Errno 36] File name too long: '/home/haoxuan/.cache/huggingface/hub/models--Salesforce--blip2-opt-6.7b/blobs/W/"4ebe4bb3f3114daf2e4cc349f24873a1175a35d7.lock'`.

However, the length of above file is only 126.
```
>>> len('/home/haoxuan/.cache/huggingface/hub/models--Salesforce--blip2-opt-6.7b/blobs/W/"4ebe4bb3f3114daf2e4cc349f24873a1175a35d7.lock')  
126
```

### Reproduction

My code is borrowed from https://github.com/Vision-CAIR/ChatCaptioner. I tried different versions of BLIP-2 and they all fail. It seems others can successfully download pre-trained BLIP2 without any problem. Would that be something wrong with my system config?

### Logs

```shell
Traceback (most recent call last):
  File "/home/haoxuan/code/PGV/blip_gpt_main.py", line 584, in <module>
    main(args)
  File "/home/haoxuan/code/PGV/blip_gpt_main.py", line 555, in main
    'OPT6.7B': Blip2('OPT6.7B', device_id=args.device_id, bit8=True),
  File "/home/haoxuan/code/PGV/blip_gpt_main.py", line 113, in __init__
    self.blip2_processor = Blip2Processor.from_pretrained(BLIP2DICT[self.tag])
  File "/dvmm-filer2/users/haoxuan/anaconda3/envs/chatcap/lib/python3.9/site-packages/transformers/processing_utils.py", line 184, in from_pretrained
    args = cls._get_arguments_from_pretrained(pretrained_model_name_or_path, **kwargs)
  File "/dvmm-filer2/users/haoxuan/anaconda3/envs/chatcap/lib/python3.9/site-packages/transformers/processing_utils.py", line 228, in _get_arguments_from_pretrained
    args.append(attribute_class.from_pretrained(pretrained_model_name_or_path, **kwargs))
  File "/dvmm-filer2/users/haoxuan/anaconda3/envs/chatcap/lib/python3.9/site-packages/transformers/models/auto/tokenization_auto.py", line 694, in from_pretrained
    return tokenizer_class.from_pretrained(pretrained_model_name_or_path, *inputs, **kwargs)
  File "/dvmm-filer2/users/haoxuan/anaconda3/envs/chatcap/lib/python3.9/site-packages/transformers/tokenization_utils_base.py", line 1770, in from_pretrained
    resolved_vocab_files[file_id] = cached_file(
  File "/dvmm-filer2/users/haoxuan/anaconda3/envs/chatcap/lib/python3.9/site-packages/transformers/utils/hub.py", line 409, in cached_file
    resolved_file = hf_hub_download(
  File "/dvmm-filer2/users/haoxuan/anaconda3/envs/chatcap/lib/python3.9/site-packages/huggingface_hub/utils/_validators.py", line 120, in _inner_fn
    return fn(*args, **kwargs)
  File "/dvmm-filer2/users/haoxuan/anaconda3/envs/chatcap/lib/python3.9/site-packages/huggingface_hub/file_download.py", line 1296, in hf_hub_download
    with FileLock(lock_path):
  File "/dvmm-filer2/users/haoxuan/anaconda3/envs/chatcap/lib/python3.9/site-packages/filelock/_api.py", line 220, in __enter__
    self.acquire()
  File "/dvmm-filer2/users/haoxuan/anaconda3/envs/chatcap/lib/python3.9/site-packages/filelock/_api.py", line 173, in acquire
    self._acquire()
  File "/dvmm-filer2/users/haoxuan/anaconda3/envs/chatcap/lib/python3.9/site-packages/filelock/_unix.py", line 35, in _acquire
    fd = os.open(self._lock_file, open_mode)
OSError: [Errno 36] File name too long: '/home/haoxuan/.cache/huggingface/hub/models--Salesforce--blip2-opt-6.7b/blobs/W/"4ebe4bb3f3114daf2e4cc349f24873a1175a35d7.lock'
```

### System info

```shell
- huggingface_hub version: 0.13.3
- Platform: Linux-4.15.0-189-generic-x86_64-with-glibc2.27
- Python version: 3.9.16
- Running in iPython ?: No
- Running in notebook ?: No
- Running in Google Colab ?: No
- Token path ?: /home/haoxuan/.cache/huggingface/token
- Has saved token ?: False
- Configured git credential helpers:
- FastAI: N/A
- Tensorflow: N/A
- Torch: 1.12.1
- Jinja2: 3.1.2
- Graphviz: N/A
- Pydot: N/A
- Pillow: 9.4.0
- hf_transfer: N/A
- ENDPOINT: https://huggingface.co
- HUGGINGFACE_HUB_CACHE: /home/haoxuan/.cache/huggingface/hub
- HUGGINGFACE_ASSETS_CACHE: /home/haoxuan/.cache/huggingface/assets
- HF_TOKEN_PATH: /home/haoxuan/.cache/huggingface/token
- HF_HUB_OFFLINE: False
- HF_HUB_DISABLE_TELEMETRY: False
- HF_HUB_DISABLE_PROGRESS_BARS: None
- HF_HUB_DISABLE_SYMLINKS_WARNING: False
- HF_HUB_DISABLE_IMPLICIT_TOKEN: False
- HF_HUB_ENABLE_HF_TRANSFER: False
```

**Comment by EthanZoneCoding** [comment]
I'm having similar issues. For me, it loads forever, until I delete the `blobs` folder in which I get
```
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\EthanPorcaro/.cache\\huggingface\\diffusers\\models--runwayml--stable-diffusion-v1-5\\blobs\\W/"a087b121ef108ea987e84c9c81f5759a81666f29.lock'
```

**Comment by EthanZoneCoding** [comment]
The fix in #1423 helped for me

**Comment by Wauplin** [comment]
Hi, thanks for reporting :) The issue has been identified and fixed server-side.

PR https://github.com/huggingface/huggingface_hub/pull/1428 will still be merged to handle this case in the future.
I'm now closing this issue but feel free to re-open if you still have issues on this matter.
